core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/gather.kt

```diff
@@ -136,6 +464,35 @@ public class Gather<T, C, K, R>(
 
 // region into
 
+/**
+ * Reshapes the columns previously selected by [gather] into two new columns:
+ * [keyColumn], containing the original column names, and [valueColumn], containing the corresponding cell values.
+ *
+ * Returns a new [DataFrame] with the reshaped structure.
+ * The original gathered columns are removed from the result,
+ * while all other columns remain unchanged —
+ * except that their values are duplicated for each generated key-value pair.
+ *
+ * Resulting key and value values can be adjusted using [mapKeys] and [mapValues], respectively.
```

Comment:
If you phrase it like this, I'd read "Resulting" as "after this function", but you mean "before calling this function", right? Maybe just leave out "Resulting", I think it's clear enough without it. Same for the functions below.